solvers/relay_constellation/mclp_teg_contact_plan/src/time_grid.py

```
@@ -0,0 +1,41 @@
+    horizon_start: datetime,
+    instant: datetime,
+    routing_step_s: int,
+) -> int:
+    """Return the grid index for an instant; raises ValueError if off-grid."""
+    if routing_step_s <= 0:
+        raise ValueError("routing_step_s must be > 0")
+    delta = instant - horizon_start
+    total_seconds = delta.total_seconds()
+    step = routing_step_s
+    idx_float = total_seconds / step
+    idx_rounded = int(round(idx_float))
```

Ruff (0.15.11)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)